modules/sources/transformers/snapshots-v3-with-events-data.ts

```diff
@@ -0,0 +1,274 @@
+import _ from 'lodash';
+import { PoolSnapshotFragment } from '../subgraphs/balancer-v3-vault/generated/types';
+import { Chain, Prisma, PrismaPoolSnapshot } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { weiToFloat } from '../../common/numbers';
+
+interface PoolStatsFromEvents {
+    id: string;
+    timestamp: number;
+    pool_id: string;
+    total_swaps: number;
+    total_join_exits: number;
+    daily_volume: number;
+    inflow: number;
+    outflow: number;
+    balance: number;
+    volume: number;
+    tvl: number;
+}
+
+interface PoolVolumeFromSwaps {
+    id: string;
+    timestamp: number;
+    pool_id: string;
+    total_swaps: number;
+    daily_volume: number;
+    volume: number;
+}
+
+interface PoolTvlFromJoinsExits {
+    id: string;
+    timestamp: number;
+    pool_id: string;
+    total_join_exits: number;
+    inflow: number;
+    outflow: number;
+    daily_balance: number;
+    tvl: number;
+}
+
+const getVolumeStatsFromSwaps = async (poolIds: string[], timestamps: number[]): Promise<PoolVolumeFromSwaps[]> => {
+    // Get data for the given pool IDs and timestamps
+    const query = Prisma.sql`
+        WITH daily_volumes AS (
+            SELECT
+                date_trunc('day', to_timestamp("blockTimestamp")) AS day,
+                "poolId" as pool_id,
+                COUNT(*) AS total_swaps,
+                SUM("valueUSD") AS daily_volume
+            FROM
+                "PrismaPoolEvent"
+            WHERE
+                type = 'SWAP'
+            GROUP BY
+                1, 2
+        )
+        SELECT
+            CONCAT(dv.pool_id, '-', EXTRACT(epoch from dv.day)::int) AS id,
+            EXTRACT(epoch from dv.day)::int AS timestamp,
+            dv.pool_id,
+            dv.total_swaps,
+            dv.daily_volume,
+            SUM(dv.daily_volume) OVER (PARTITION BY dv.pool_id ORDER BY dv.day ASC) AS volume
+        FROM
+            daily_volumes dv
+        WHERE
+            dv.pool_id IN (${Prisma.join(poolIds, ',')})
+            AND EXTRACT(epoch from dv.day)::int::text IN (${Prisma.join(timestamps.map(String), ',')})
+    `;
+
+    const poolVolumes = await prisma.$queryRaw<PoolVolumeFromSwaps[]>(query);
```

Comment:
sorry but to me, this is just magic. No idea what that does. I wont be able to maintain or debug this. Is there a way to do that in TS instead of SQL? 
I would assume, in general, this needs to be done for many pools for exactly one day.